### PR TITLE
Feature/student signup

### PR DESCRIPTION
**Pull request by gauravkeerthi**
Completed the skeleton flow for the room participants, including updating the room database and creating a global context

### CODE REVIEW

File: src/models/roomModel.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export const updateRoomParticipantsDb = async (id, userId) => {
+  
+    const roomRef = firestore.collection('rooms').doc(id)
+    roomRef.get().then((doc) => {
+      if (doc.exists) {
+        if (doc.data().numParticipants == null) {
+          roomRef.set({ // or should it be set
+            // numParticipants: 1,
+            participants: [userId]
```
Comment (halfwhole) [inline src/models/roomModel.js:31]: Could we change the `participants` here to `participantIds`?

File: src/models/roomModel.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        if (doc.data().participants.includes(userId)) {
+          console.log("User already exists in room");
+          return null
+        }
+        else {
+          roomRef.update(
+            {
+              // numParticipants: doc.data().numParticipants + 1,
+              participants: [...doc.data().participants,userId]
```
Comment (halfwhole) [inline src/models/roomModel.js:42]: Here as well!

File: src/models/roomModel.js (hunk 1 above)
Comment (gauravkeerthi) [reply src/models/roomModel.js:31]: Done

File: src/models/roomModel.js (hunk 2 above)
Comment (gauravkeerthi) [reply src/models/roomModel.js:42]: done

**Pull request merged**